test/test_jsinterp.py

```diff
@@ -1,117 +1,89 @@
 #!/usr/bin/env python
 
+# """
+# see: `js2tests`
+# """
+
 from __future__ import unicode_literals
 
 # Allow direct execution
 import os
 import sys
-import unittest
+import logging
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
 sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
-from youtube_dl.jsinterp import JSInterpreter
+from youtube_dl.jsinterp2 import JSInterpreter
+from .js2tests import gettestcases
 
+defs = gettestcases()
+# set level to logging.DEBUG to see messages about missing assertions
+logging.basicConfig(stream=sys.stderr, level=logging.WARNING)
 
-class TestJSInterpreter(unittest.TestCase):
```

Comment:
Don't remove tests for the original JSInterpreter. Before jsinterp2 replaces jsinterp, tests should be there to guarantee nothing gets broken by accident.

Reply:
As right now both `test_jsinterp_orig` and `test_jsinterp` exist later is using my suite, but should do the same.

Reply:
I'd prefer leaving `test_jsinterp.py` as-is and add new tests in `test_jsinterp2.py`, so that the test file name matches the interpreter module name.

Reply:
Here's a different idea: new tests: `test_js2test_legacy.py`, `test_js2test_jsinterp2.py`, `test_js2test_jsinterp2_parse.py`; and the original: `test_jsinterp.py` kept untouched.
The tests `test_js2test_legacy.py` performs tests on the original `jsinterp.py`, but using the new test suite.
I'm not a big fan of these names, but this is what I could come up with.
On one hand there's a suggestion to the method the testcases are generated, on the other they're bit convoluted.